sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfigTranslator.java

```diff
@@ -117,22 +119,26 @@ private static BigtableDataSettings.Builder buildBigtableDataSettings(
       dataBuilder.setAppProfileId(Objects.requireNonNull(config.getAppProfileId().get()));
     }
 
-    if (((GcpOptions) pipelineOptions).getGcpCredential() != null) {
-      dataBuilder
-          .stubSettings()
-          .setCredentialsProvider(
-              FixedCredentialsProvider.create(((GcpOptions) pipelineOptions).getGcpCredential()));
-    }
-
-    if (config.getCredentialFactory() != null) {
-      CredentialFactory credentialFactory = config.getCredentialFactory();
-      try {
+    // Skip resetting the credentials if it's connected to an emulator
+    if (!emulator) {
```

Comment:
Here the code path is made different intentionally for test and real use cases. Does this cause the if {} clause no longer covered by unit test where it was intended to be?

Reply:
This is still covered by `testUsingPipelineOptionsCredential` and `testUsingCredentialsFromBigtableOptions` in `BigtableConfigTranslatorTest` to make sure the credentials are updated correctly. The emulator test case is tested by the sql tests.